cost-analyzer/templates/NOTES.txt

```diff
@@ -1,5 +1,6 @@
 --------------------------------------------------
 {{- include "kubecostV2-preconditions" . -}}
+{{- include "cloudIntegrationSecretMutualExclusivityCheck" . -}}
```

Comment:
more of a philosophical discussion here, but do you know when NOTES is rendered, chip? I can't find it in https://github.com/helm/helm/blob/release-3.0/pkg/releaseutil/kind_sorter.go#L27 . I would hate to have it rendered at the end and we blow the install up after the bad stuff has already been sent to API server

Reply:
All the resources are rendered internally before being parcelled out to the API server in the order described [here](https://helm.sh/docs/intro/using_helm/). If a failure is encountered, no resources will be submitted.

```sh
$ k get ns
NAME              STATUS   AGE
kube-system       Active   41h
default           Active   41h
kube-public       Active   41h
kube-node-lease   Active   41h

$ helm install kubecost -n kubecost cost-analyzer/ --create-namespace
Error: INSTALLATION FAILED: execution error at (cost-analyzer/templates/NOTES.txt:3:4): cloudIntegrationSecret and cloudIntegrationJSON are mutually exclusive. Please specify only one.

$ k get ns
NAME              STATUS   AGE
kube-system       Active   41h
default           Active   41h
kube-public       Active   41h
kube-node-lease   Active   41h
```